docs/topics/deployment/docker.rst

```diff
@@ -1,40 +1,192 @@
 Docker
-===========
+======
 
-This uses the default python container https://hub.docker.com/_/python
+Docker is a containerization platform that allows you to package your application and all its dependencies together.
+It is useful for creating consistent environments for your application to run in, irrespective of the host system
+and its own configuration or dependencies - which is especially helpful in preventing dependency conflicts.
+
+This guide uses the `Docker official Python container <https://hub.docker.com/_/python>`_ as a base image.
+
+Use When
+--------
+
+Docker is ideal for deploying Python web applications in scenarios where:
+
+- **Isolation:** You require a consistent, isolated environment for your application, independent of the host system.
+- **Scalability:** Your application needs to be easily scaled up or down based on demand.
+- **Portability:** The need to run your application consistently across different environments (development, testing, production) is crucial.
+- **Microservices Architecture:** You are adopting a microservices architecture, where each service can be containerized and managed independently.
+- **Continuous Integration/Continuous Deployment (CI/CD):** You are implementing CI/CD pipelines, and Docker facilitates the building, testing, and deployment of applications.
+- **Dependency Management:** Ensuring that your application has all its dependencies bundled together without conflicts with other applications.
+
+Alternatives
+~~~~~~~~~~~~
+
+For different deployment scenarios, consider these alternatives:
+
+- :doc:`NGINX Unit <nginx-unit>`:
```

Comment:
NGINX unit as an alternative to docker? I think that point needs a bit more explanation 👀